xbmc/pvr/PVRDatabase.cpp

```diff
@@ -488,36 +490,17 @@ int CPVRDatabase::Get(CPVRChannelGroup &group)
 
 bool CPVRDatabase::PersistChannels(CPVRChannelGroup &group)
 {
-  bool bReturn(true);
+  for (auto &member : group.m_members) {
+    const CPVRChannelPtr &channel = member.second.channel;
```

Comment:
I just want to understand if this is a mistake or intentional. A ref to a smart pointer makes no sense. Why don't you want the smart pointer increment its internal ref counter?